### PR TITLE
wdpost: fix post with terminated sectors

### PR DESCRIPTION
**Pull request by magik6k**
Should fix https://github.com/filecoin-project/lotus/issues/3808, https://github.com/filecoin-project/lotus/issues/3788

### CODE REVIEW

File: storage/wdpost_run.go [hunk 1]
```diff
@@ -467,7 +467,7 @@ func (s *WindowPoStScheduler) runPost(ctx context.Context, di dline.Info, ts *ty
 
 				skipCount += sc
 
-				ssi, err := s.sectorsForProof(ctx, good, partition.Sectors, ts)
+				ssi, err := s.sectorsForProof(ctx, good, toProve, ts)
```
Comment (magik6k) [inline storage/wdpost_run.go:470]: `partition.Sectors` included terminated sectors.. actor code expects `Sectors - Terminated - (Faults - Recoveries)`

**Comment by magik6k** [comment]
Example impacted partition:
```
./lotus chain get --tipset @54182 /pstate/@Ha:t07919/1/11/0/0/0/@A:0

[
        "7CjaypWOkotKU1Mc0alUKoqDh6K8F5FMdBVspCuOI3U8x1GU2teVdi0W8Si1MqHcETvVeqgqyq9YTqVyRdEUKVK1Zi8RV/qVWkSyk46dCjFRqWTDcUph6tYy1Oy1mpnoxjZA6Wxmo9RRrCsZSVzpdkdRVhxpZ01RWgdCkdudUBQtRVGczrWeUvGagVhxV3aKzh1JuXJHq9yp/H/nSkuxblVairNSkq46v1PpfO9aRbF2nBVHWekoyjs7O0rlSu9KR1EUpVKpVGqOpFxRvnLlTkVZuVKSOq/0JKlSqlRWnI7iKFKn21upKEpHUpz/zneurCjOiqJ0dlZecZRbSmXF6XRe+sqK9J1KpXTtu71u81q9FYk7TEhOPRTdzqUdZcUqXan1Kq1LlY5SUZSdLyk7v3NHcpSS4imO0lE6tepVRXFW7pQqJUXpOErniiOtKIpSUZQ7lUsr3hVFcpxrJWdFUkpXFKWirO14lc6lYufKv2IpjqLUateUV5ROq1JaWnlr6YpUkZRORXEURVGcSiRKf8VxlKkI",
        "",
        "",
        "YCw=", # Terminations
        {
                "/": "bafy2bzacedrpuahiltmzjoet3kqlmk3bpoj6ky2umykg65kpy5a7v54q5qsky"
        },
        {
                "/": "bafy2bzacedswlcz5ddgqnyo3sak3jmhmkxashisnlpq6ujgyhe4mlobzpnhs6"
        },
        [
                "AElgAAAAAA==",
                "AElgAAAAAA=="
        ],
        [
                "",
                ""
        ],
        [
                "",
                ""
        ]
]

```

```
./lotus-shed bitfield intersect "YCw=" "7CjaypWOkotKU1Mc0alUKoqDh6K8F5FMdBVspCuOI3U8x1GU2teVdi0W8Si1MqHcETvVeqgqyq9YTqVyRdEUKVK1Zi8RV/qVWkSyk46dCjFRqWTDcUph6tYy1Oy1mpnoxjZA6Wxmo9RRrCsZSVzpdkdRVhxpZ01RWgdCkdudUBQtRVGczrWeUvGagVhxV3aKzh1JuXJHq9yp/H/nSkuxblVairNSkq46v1PpfO9aRbF2nBVHWekoyjs7O0rlSu9KR1EUpVKpVGqOpFxRvnLlTkVZuVKSOq/0JKlSqlRWnI7iKFKn21upKEpHUpz/zneurCjOiqJ0dlZecZRbSmXF6XRe+sqK9J1KpXTtu71u81q9FYk7TEhOPRTdzqUdZcUqXan1Kq1LlY5SUZSdLyk7v3NHcpSS4imO0lE6tepVRXFW7pQqJUXpOErniiOtKIpSUZQ7lUsr3hVFcpxrJWdFUkpXFKWirO14lc6lYufKv2IpjqLUateUV5ROq1JaWnlr6YpUkZRORXEURVGcSiRKf8VxlKkI"

YCw=
```

File: storage/wdpost_run.go (hunk 1 above)
Comment (Stebalien) [reply storage/wdpost_run.go:470]: I'm not sure what was wrong here, but this isn't the right fix. We _should_ include terminated sectors in this bitfield. `sectorsForProof` will (or should) replace these terminated sectors (really, all sectors _not_ included in "good") with the first "good" sector.

See https://github.com/filecoin-project/specs-actors/blob/d48c45d15f1d79b06bc27c5be326a9b88c1fae39/actors/builtin/miner/deadline_state.go#L961

**Review by Stebalien — CHANGES_REQUESTED** [review]

**Comment by magik6k** [comment]
This is wrong

**Pull request closed without merging**